yarn-project/p2p/src/services/libp2p/libp2p_service.ts

```diff
@@ -360,7 +360,12 @@ export class LibP2PService<T extends P2PClientType = P2PClientType.Full> extends
 
     // Update gossipsub score params
     node.services.pubsub.score.params.appSpecificWeight = 10;
-    node.services.pubsub.score.params.appSpecificScore = (peerId: string) => peerManager.getPeerScore(peerId);
+    node.services.pubsub.score.params.appSpecificScore = (peerId: string) => {
+      if (peerManager.shouldDisableP2PGossip(peerId)) {
+        return -Infinity;
+      }
+      return peerManager.getPeerScore(peerId);
+    };
```

Comment:
```suggestion
    node.services.pubsub.score.params.appSpecificScore = (peerId: string) => peerManager.shouldDisableP2PGossip(peerId) ? -Infinity : peerManager.getPeerScore(peerId);
```
Just a style preference. No need to change though.